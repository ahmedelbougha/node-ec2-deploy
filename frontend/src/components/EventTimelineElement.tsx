import { VerticalTimelineElement } from 'react-vertical-timeline-component';
import CheckCircle from '@material-ui/icons/CheckCircle';

import { EventTypes } from '../utils/constants';
import { EventTimelineElementProps } from './EventTimelineElement.d';

export default function EventTimelineElement({
  event,
}: EventTimelineElementProps) {
  return (
    <VerticalTimelineElement
      className="vertical-timeline-element--work"
      contentStyle={{
        background: EventTypes[event.event_type]['background'],
        color: EventTypes[event.event_type]['color'],
      }}
      contentArrowStyle={{
        borderRight: `7px solid ${EventTypes[event.event_type]['background']}`,
      }}
      iconStyle={{
        background: EventTypes[event.event_type]['background'],
        color: EventTypes[event.event_type]['color'],
      }}
      icon={<CheckCircle />}
    >
      <h3 className="vertical-timeline-element-title">
        {event.event_type.replaceAll('_', ' ').toUpperCase()}
      </h3>
      <h4 className="vertical-timeline-element-subtitle">
        {new Date(event.timestamp).toDateString()}
      </h4>
      <div>
        {EventTypes[event.event_type]['extraData'].map(
          // please check the EventTypes constant for extraData array of keys
          // keys vary based on the event type
          (dataKey: string, index): JSX.Element => (
            <p key={dataKey + index}>
              {(event.payload[dataKey] as string) || ''}
            </p>
          )
        )}
      </div>
    </VerticalTimelineElement>
  );
}
