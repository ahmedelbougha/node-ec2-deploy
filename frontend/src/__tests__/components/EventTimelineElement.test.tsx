import renderer from 'react-test-renderer';
import { EventTimelineElementProps } from '../../components/EventTimelineElement.d';
import EventTimelineElement from '../../components/EventTimelineElement';

const timelineElementProps: EventTimelineElementProps = {
  event: {
    id: 'c6626ebd-dc8e-4b4e-b25c-170c1338df4a',
    care_recipient_id: 'df50cac5-293c-490d-a06c-ee26796f850d',
    event_type: 'task_completed',
    timestamp: '2019-05-12T18:54:09.116Z',
    payload: {
      id: 'c6626ebd-dc8e-4b4e-b25c-170c1338df4a',
      visit_id: '5cd753f0-8b66-f8a8-43f7-330f62a3e1d6',
      timestamp: '2019-05-12T18:54:09.116Z',
      event_type: 'task_completed',
      caregiver_id: 'f7a00df5-bbc4-4ad7-9918-c07e16e709f6',
      task_instance_id:
        'dHw2YzBjZGViMy1mNzhlLTQwNTUtYmRlMy1kMmVmYTdlMmFmNjZ8MjAxOS0wNS0xMlQxOTowMDowMC4wMDBafEVWRU5JTkc=',
      task_schedule_id: '6c0cdeb3-f78e-4055-bde3-d2efa7e2af66',
      care_recipient_id: 'df50cac5-293c-490d-a06c-ee26796f850d',
      task_definition_id: 'f71f00c2-ed7e-4366-a2be-d2abc6eb939b',
      task_schedule_note: 'Please change my pad and give me a freshen up',
      task_definition_description: 'Assist to change incontinence pants',
    },
  },
};

describe('Event Timeline Element', () => {
  it('renders', () => {
    const component = renderer.create(
      <EventTimelineElement {...timelineElementProps} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
