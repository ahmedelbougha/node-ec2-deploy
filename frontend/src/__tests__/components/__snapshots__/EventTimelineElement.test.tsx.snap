// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Timeline Element renders 1`] = `
<div
  className="vertical-timeline-element--work vertical-timeline-element"
  id=""
  style={null}
>
  <span
    className="vertical-timeline-element-icon is-hidden"
    onClick={null}
    style={
      Object {
        "background": "#28b5bd",
        "color": "#fff",
      }
    }
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
  </span>
  <div
    className="vertical-timeline-element-content is-hidden"
    onClick={null}
    style={
      Object {
        "background": "#28b5bd",
        "color": "#fff",
      }
    }
  >
    <div
      className="vertical-timeline-element-content-arrow"
      style={
        Object {
          "borderRight": "7px solid #28b5bd",
        }
      }
    />
    <h3
      className="vertical-timeline-element-title"
    >
      TASK COMPLETED
    </h3>
    <h4
      className="vertical-timeline-element-subtitle"
    >
      Sun May 12 2019
    </h4>
    <div>
      <p>
        Please change my pad and give me a freshen up
      </p>
      <p>
        Assist to change incontinence pants
      </p>
    </div>
    <span
      className="vertical-timeline-element-date"
    >
      
    </span>
  </div>
</div>
`;
